elks/arch/i86/kernel/system.c

```diff
@@ -130,4 +144,14 @@ void apm_shutdown_now(void)
         "apm_error:\n\t"
     );
 #endif
+#ifdef CONFIG_ARCH_SWAN
+    /* Disable interrupts */
+    outb(0x00, 0xB2);
+    outb(0xFF, 0xB6);
+    clr_irq();
+    /* Request poweroff */
+    outb(1, 0x62);
+    /* Halt CPU */
+    while(1) asm("hlt");
+#endif
```

Comment:
I've been aware for some time that hard_reset_now and apm_shutdown_now need to get pulled out into specific architecture files. Lets leave this in for now, I've made a note of it.